Sources/Core/Tracker/Tracker.swift

```diff
@@ -345,17 +345,19 @@ class Tracker: NSObject {
 
     private func checkInstall() {
         if installEvent {
-            let installTracker = InstallTracker()
-            let previousTimestamp = installTracker.previousInstallTimestamp
-            installTracker.clearPreviousInstallTimestamp()
-            if !installTracker.isNewInstall && previousTimestamp == nil {
-                return
+            DispatchQueue.global(qos: .default).async { [weak self] in
```

Comment:
In the issue you say...
> Currently, the application_install event is being generated before the additional callback configuration has been processed.
>
> This works as expected in the Snowplow Android tracker.

I wonder way it works on Android. Your current solution put this block in a queue for the execution. I think it works because it's queued after the fetcher that download and apply the remote configuration. I don't think this completely solve the problem as you can have race conditions where this block is executed before the fetching ending up with the same issue we are trying to solve. Happy to be proved wrong btw.

This solution can be a temporary improvement but, in my opinion, this is a non trivial problem.
We should be sure that the install event is tracked with the right configuration which is the one fetched by remote configuration. At the moment, there is no way for the Tracker class to know if that Tracker instance comes from a default configuration or a remote one. Instead, this is an important information for install_event or other events. In this case we don't want to fire it until we have applied the remote configuration. So, in theory, we should make the Tracker class aware of our intention. Is it a tracker fully configured or we are waiting to fetch a remote config? If the Tracker is waiting for a remote config, it shouldn't fire any install_event until a timeout (defined somewhere) trigger the event.
That would exclude any possible race condition IMO.